launch/dynamo-run/src/subprocess/sglang_inc.py

```diff
@@ -54,12 +59,19 @@ def __init__(self, engine):
         self.engine_client = engine
 
     async def generate(self, request):
-        # print(f"Received request: {request}")
-        sampling_params = {
-            "temperature": request["sampling_options"]["temperature"],
-            # sglang defaults this to 128
-            "max_new_tokens": request["stop_conditions"]["max_tokens"],
-        }
+        logging.debug("Generate start here")
+
+        sampling_params = {}
+        for key, value in request["sampling_options"].items():
+            if value:
+                # TODO: Do these always match? Maybe allow-list the fields that do match
+                sampling_params[key] = valure
```

Comment:
Typo? Did this code path ever get hit?
```suggestion
                sampling_params[key] = value
```

Reply:
I could have sworn I tested it. But not, apparently:
```
2025-05-07T20:59:25.131Z DEBUG dynamo_run::subprocess: 
2025-05-07T20:59:25.131Z DEBUG dynamo_run::subprocess: Traceback (most recent call last):
2025-05-07T20:59:25.131Z DEBUG dynamo_run::subprocess:   File "/tmp/.tmpBGIoKX", line 68, in generate
2025-05-07T20:59:25.131Z DEBUG dynamo_run::subprocess:     sampling_params[key] = valure
2025-05-07T20:59:25.131Z DEBUG dynamo_run::subprocess:                
```

:1st_place_medal: for bug finding.